### PR TITLE
timeshift: new, 24.06.5

### PR DESCRIPTION
**Pull request by miwu04**
Topic Description
-----------------

- timeshift: new, 24.06.5

Package(s) Affected
-------------------

- timeshift: 24.06.5

Security Update?
----------------

No

Build Order
-----------

```
#buildit timeshift
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by miwu04** [comment]
342ad0a962f9e2a2f805769c00225b7d

File: app-utils/timeshift/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+PKGSEC=utils
+PKGDES="System Restore Utility"
+BUILDDEP="meson help2man gettext vala vte libgee json-glib xapps"
+PKGDEP="at-spi2-core glibc cairo gdk-pixbuf libgee glib gnutls gtk-3 json-glib pango pcre2 vte x11-lib zlib rsync psmisc"
+
+PKGPROV=timeshift-btrfs
+
+ABTYPE=meson
+
+FAIL_ARCH="!(amd64|arm64)"
```
Comment (eatradish) [inline app-utils/timeshift/autobuild/defines]: Why FAIL_ARCH?

File: app-utils/timeshift/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,11 @@
+PKGNAME=timeshift
+PKGSEC=utils
+PKGDES="System Restore Utility"
+BUILDDEP="meson help2man gettext vala vte libgee json-glib xapps"
+PKGDEP="at-spi2-core glibc cairo gdk-pixbuf libgee glib gnutls gtk-3 json-glib pango pcre2 vte x11-lib zlib rsync psmisc"
```
Comment (eatradish) [inline app-utils/timeshift/autobuild/defines:5]: dependencies from `ldd` output:

```
atk
at-spi2-atk
at-spi2-core
brotli
bzip2
cairo
cryptsetup
dbus
expat
fontconfig
freetype
fribidi
gcc
gcc-runtime
gdk-pixbuf
glib
glibc
gmp
gnutls
graphite
gtk-3
harfbuzz
icu
json-c
json-glib
libcap
libdatrie
libepoxy
libffi
libgee
libidn2
libjpeg-turbo
libpng
libtasn1
libthai
libtiff
libunistring
libwebp
libxau
libxcb
libxdmcp
libxkbcommon
lvm2
nettle
openssl
p11-kit
pango
pcre2
pixman
systemd
util-linux-runtime
vte
wayland
x11-lib
xz
zlib
zstd
```

Step:

```bash
for i in `ldd /usr/bin/timeshift | choose 2`; do oma provides --no-progress $i >> 1; done
cat 1 | choose -f ":" 0 | sort | uniq
```

**Review by eatradish — CHANGES_REQUESTED** [review]

File: app-utils/timeshift/autobuild/defines (hunk 2 above)
Comment (eatradish) [reply app-utils/timeshift/autobuild/defines:5]: A better way:

```bash
for i in `readelf -d /usr/bin/timeshift | grep NEEDED | choose 2 | choose -f "：" 1 | choose -f "\[" 0 | choose -f "\]" 0`; do oma provides --no-progress $i >> 1; done
cat 1 | choose -f ":" 0 | sort | uniq
```

File: app-utils/timeshift/autobuild/defines (hunk 1 above)
Comment (eatradish) [reply app-utils/timeshift/autobuild/defines]:
```bash
for i in `LANG=C readelf -d /usr/bin/timeshift | grep NEEDED | choose -1 | choose -f "\[" 0 | choose -f "\]" 0`; do oma provides --no-progress $i >> 1; done
cat 1 | choose -f ":" 0 | sort | uniq
```

Result:

```
gcc+cross-mips64r6el
gdk-pixbuf
gdk-pixbuf+32
glib
glib+32
glibc
glibc+32
gtk-3
gtk-3+32
json-glib
json-glib-dbg
libgee
vte
```

**Review by eatradish — CHANGES_REQUESTED** [review]
```
saki@MagMilk [ ~ ] $ timeshift-gtk
timeshift-gtk: error while loading shared libraries: libxapp.so.1: cannot open shared object file: No such file or directory
```

**Comment by miwu04** [comment]
> ```
> saki@MagMilk [ ~ ] $ timeshift-gtk
> timeshift-gtk: error while loading shared libraries: libxapp.so.1: cannot open shared object file: No such file or directory
> ```
done

**Review by eatradish — APPROVED** [review]

**Pull request merged**